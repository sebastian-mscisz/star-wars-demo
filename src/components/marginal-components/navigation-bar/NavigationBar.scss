@import 'src/variables';

.sw-nav {
    font-size: 1.7rem;
    font-weight: 700;

    @media screen and (max-width: $small-bp) {
        font-size: 1.2rem;
    }

    &__links {
        list-style: none;
        padding-left: 30px;

        @media screen and (max-width: $small-bp) {
            display: flex;
            padding-left: 0px;
            justify-content: space-evenly;
        }

        &__item {
            padding-top: 10px;

            @media screen and (max-width: $small-bp) {
                padding-top: 0px;
            }

            &__link {
                text-decoration: none;
                color: #b5b7b7;
                transition: all 0.3s linear;

                &--active {
                    text-decoration: underline;
                    color: #f8f8f8;
                }

                &:hover {
                    color: #f8f8f8;
                }
            }
        }
    }
}
