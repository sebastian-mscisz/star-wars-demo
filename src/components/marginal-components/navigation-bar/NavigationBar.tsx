import { charactersPagePath, locationsPagePath, vehiclesPagePath } from 'src/common';
import { NavLink } from 'react-router-dom';
import './NavigationBar.scss';

const navigationItems = [
    { name: 'Characters', path: charactersPagePath() },
    { name: 'Locations', path: locationsPagePath() },
    { name: 'Vehicles', path: vehiclesPagePath() },
];

export const NavigationBar: React.FC = () => {
    const navlinks = navigationItems.map((item, i) => (
        <li className="sw-nav__links__item" key={i}>
            <NavLink
                className={({ isActive }) =>
                    isActive
                        ? 'sw-nav__links__item__link sw-nav__links__item__link--active'
                        : 'sw-nav__links__item__link'
                }
                key={item.name}
                to={item.path}
            >
                {item.name}
            </NavLink>
        </li>
    ));

    return (
        <div className="sw-nav">
            <ul className="sw-nav__links">{navlinks}</ul>
        </div>
    );
};
