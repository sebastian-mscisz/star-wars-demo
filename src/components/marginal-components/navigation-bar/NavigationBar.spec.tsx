import { BrowserRouter } from 'react-router-dom';
import { NavigationBar } from './NavigationBar';
import { render } from '@testing-library/react';

describe('Navigation bar', () => {
    it('should render', () => {
        const { asFragment } = render(<NavigationBar />, { wrapper: BrowserRouter });
        expect(asFragment()).toMatchSnapshot();
    });
});
