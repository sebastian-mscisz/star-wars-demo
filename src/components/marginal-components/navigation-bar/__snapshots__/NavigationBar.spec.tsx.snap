// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar should render 1`] = `
<DocumentFragment>
  <div
    class="sw-nav"
  >
    <ul
      class="sw-nav__links"
    >
      <li
        class="sw-nav__links__item"
      >
        <a
          class="sw-nav__links__item__link"
          href="/characters"
        >
          Characters
        </a>
      </li>
      <li
        class="sw-nav__links__item"
      >
        <a
          class="sw-nav__links__item__link"
          href="/locations"
        >
          Locations
        </a>
      </li>
      <li
        class="sw-nav__links__item"
      >
        <a
          class="sw-nav__links__item__link"
          href="/vehicles"
        >
          Vehicles
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
